internal/blueprint/repository_customizations.go

```diff
@@ -326,3 +330,84 @@ func validateCustomRepository(repo *RepositoryCustomization) error {
 	}
 	return nil
 }
+
+// DeepCopy returns a deep copy of the given slice of RepositoryCustomization
+// Since we are replacing the gpgkeys of the customization with the file path
+// we don't want to modify the original slice.
+func DeepCopyCustomRepositories(repos []RepositoryCustomization) []RepositoryCustomization {
+	clonedRepos := make([]RepositoryCustomization, len(repos))
+	copy(clonedRepos, repos)
+	for idx, repo := range clonedRepos {
+		// copy the baseurls
+		baseurls := make([]string, len(repo.BaseURLs))
+		copy(baseurls, repo.BaseURLs)
+		clonedRepos[idx].BaseURLs = baseurls
+
+		// copy the gpgkeys
+		gpgkeys := make([]string, len(repo.GPGKeys))
+		copy(gpgkeys, repo.GPGKeys)
+		clonedRepos[idx].GPGKeys = gpgkeys
+	}
+	return clonedRepos
+}
+
+func GetInlineGPGKeysFromCustomRepositories(repos []RepositoryCustomization) ([]*fsnode.File, error) {
+	files := []*fsnode.File{}
+	for _, repo := range repos {
+		for keyIdx, gpgkey := range repo.GPGKeys {
+			// check if the gpgkey is an inline file
+			if _, ok := url.ParseRequestURI(gpgkey); ok != nil {
+				// create the file path
+				path := fmt.Sprintf("/etc/pki/rpm-gpg/RPM-GPG-KEY-%s-%d", repo.Id, keyIdx)
+				// replace the gpgkey with the file path
+				repo.GPGKeys[keyIdx] = fmt.Sprintf("file://%s", path)
+				// create the fsnode for the gpgkey file
+				file, err := fsnode.NewFile(path, common.ToPtr(os.FileMode(0777)), "root", "root", []byte(gpgkey))
+				if err != nil {
+					return nil, err
+				}
+				files = append(files, file)
+			}
+		}
+	}
+	return files, nil
+}
+
+func RepoCustomizationsToRepoConfig(repos []RepositoryCustomization) map[string][]rpmmd.RepoConfig {
+	if len(repos) == 0 {
+		return nil
+	}
+
+	m := make(map[string][]rpmmd.RepoConfig)
```

Comment:
nitpick: you should probably pass `len(repos)` as the initial capacity of the map.